test/compiler-finder-tests.js

```diff
@@ -30,37 +30,34 @@ const _ = require('underscore-node');
 chai.use(chaiAsPromised);
 const should = chai.should();
 
-const props = (key, deflt) => deflt;
-const propsL = (lang, key, def) => props(key, def);
-const propsAT = (langs, f, key, def) => {
-    let response = {};
-    _.each(langs, lang => {
-        response[langs.id] = f(propsL(lang.id, key, def));
-    });
-    return response;
+const languages = {
+    'a-lang': {
+        id: 'a-lang'
+    }
 };
 
+const props = (key, def) => key === "compilers" ? "goodCompiler:&badCompiler" : def;
+
+const compilerProps = (langs, key, defaultValue, transform) => {
```

Comment:
Why is this repeated from whats in app.js?

Reply:
I wanted to have the function be on its own file, but ad it depends upon a global variable on the app.js, I decided to to just c&p tbe function but with a slightly different implementation (This always calls props, and the one in app.js calls propsFor('langId')) I guess I can move it to a file and add some state to it? Probably the wise thing to do, yeah.

Reply:
Having behaviour like this duplicated is asking to introduce bugs that the tests will miss because well its not being tested :D

Reply:
I've moved it to its own class, and updated all the relevant code. Let me know what you think :)